### DISCUSSION TITLE
How to change Avalonia application shutdown mode

### DISCUSSION THREAD

**Discussion by Aswathips** [AvaloniaUI/Avalonia#7696] [Q&A] [answered]
Hello, I'am new to avalonia . Iam trying to develop an desktop application using avalonia. Now I want to change the application shutdown mode. The default shutdown mode is OnLastWindowClose , I want to change it to OnExplicitShutdown. But I don't know how to do it. I tried to set the shutdown mode in App.axaml file. But that is not working . 
Anyone please help me. How can I do it . Which part of the code have to change.

**Comment by night815** [comment, marked as answer]
I was just having the same issue and solved it by programmically setting the ShutdownMode in the OnFrameworkInitializationCompleted method.

```` 
public override void OnFrameworkInitializationCompleted()
{
    if (ApplicationLifetime is IClassicDesktopStyleApplicationLifetime desktop)
    {
        desktop.ShutdownMode = Avalonia.Controls.ShutdownMode.OnExplicitShutdown;
    }
}
````

**Reply by Aswathips** [reply]
Thank you very much, that's exactly what I want!
